prow/plugins/config.go

```diff
@@ -1766,3 +1767,40 @@ type Override struct {
 	// members of which are allowed to override contexts
 	AllowedGitHubTeams map[string][]string `json:"allowed_github_teams,omitempty"`
 }
+
+func (c *Configuration) mergeFrom(other *Configuration) error {
+	var errs []error
+	if diff := cmp.Diff(other, &Configuration{Plugins: other.Plugins}); diff != "" {
+		errs = append(errs, fmt.Errorf("supplemental plugin configuration has config that doesn't support merging: %s", diff))
+	}
+
+	if c.Plugins == nil {
+		c.Plugins = Plugins{}
+	}
+	if err := c.Plugins.mergeFrom(&other.Plugins); err != nil {
+		errs = append(errs, fmt.Errorf("failed to merge .plugins from supplemental config: %w", err))
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func (p *Plugins) mergeFrom(other *Plugins) error {
+	if other == nil {
```

Comment:
Doesn't the `for` implicitly handle this?

Reply:
`other` can be nil and we dererence it in the loop, which will panic if its nil